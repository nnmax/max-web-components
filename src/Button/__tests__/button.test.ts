import '..'

describe('Testing <max-button />', () => {
  it('custom elements should be defined', () => {
    const button = window.customElements.get('max-button')
    expect(button).toBeDefined()
  })

  // TODO: await for JSDOM support `HTMLElement.attachInternals()` API
  // https://github.com/jsdom/jsdom/issues/3444
  it.skip.each([
    { attribute: 'variant', value: 'contained' },
    { attribute: 'variant', value: 'outlined' },
    { attribute: 'variant', value: 'text' },
    { attribute: 'color', value: 'primary' },
    { attribute: 'color', value: 'secondary' },
    { attribute: 'type', value: 'button' },
    { attribute: 'type', value: 'submit' },
    { attribute: 'type', value: 'reset' },
    { attribute: 'disabled', value: '' },
    { attribute: 'role', value: 'menuitem' },
    { attribute: 'tabindex', value: '-1' },
  ])('should render a button with $attribute is `$value`', ({ attribute, value }) => {
    document.body.innerHTML = `
      <max-button ${attribute}="${value}">Button</max-button>
    `

    expect(document.body.innerHTML).toMatchSnapshot()
  })
})
