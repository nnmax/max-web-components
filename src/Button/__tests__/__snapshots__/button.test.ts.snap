// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <max-button /> should render a button with aria-disabled is \`\` 1`] = `
"
      <max-button aria-disabled="">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with color is \`primary\` 1`] = `
"
      <max-button color="primary">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with color is \`secondary\` 1`] = `
"
      <max-button color="secondary">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with role is \`menuitem\` 1`] = `
"
      <max-button role="menuitem">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with tabindex is \`-1\` 1`] = `
"
      <max-button tabindex="-1">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with type is \`button\` 1`] = `
"
      <max-button type="button">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with type is \`reset\` 1`] = `
"
      <max-button type="reset">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with type is \`submit\` 1`] = `
"
      <max-button type="submit">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with variant is \`contained\` 1`] = `
"
      <max-button variant="contained">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with variant is \`outlined\` 1`] = `
"
      <max-button variant="outlined">Button</max-button>
    "
`;

exports[`Testing <max-button /> should render a button with variant is \`text\` 1`] = `
"
      <max-button variant="text">Button</max-button>
    "
`;
